src/Engine/Physics/TriggerObserver.cpp

```diff
@@ -0,0 +1,112 @@
+#include <btBulletDynamicsCommon.h>
+#include <cassert>
+#include "TriggerObserver.hpp"
+
+namespace Physics {
+    TriggerObserver::TriggerObserver(btRigidBody& body)
+    : btCollisionWorld::ContactResultCallback(), rigidBody(body) {
+
+    }
+
+    btCollisionObject* TriggerObserver::GetBulletCollisionObject() {
+        return &rigidBody;
+    }
+
+    TriggerObserver::IntersectionPhase TriggerObserver::GetPhase() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetPhase' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)